perses/rjmc/topology_proposal.py

```diff
@@ -1470,6 +1474,98 @@ def generate_oemol_from_pdb_template(pdbfile):
             atom.SetName(name_without_spaces)
         return current_oemol
 
+    @staticmethod
+    def _get_charge_difference(current_resname, new_resname):
+        """
+        return the charge of the new res - charge old res
+
+        Parameters
+        ----------
+        current_resname : str
+            three letter identifier for original residue
+        new_resnme : str
+            three letter identifier for new residue
+
+        Returns
+        -------
+        chargediff : int
+            charge(new_res) - charge(old_res)
+        """
+        assert new_resname in PolymerProposalEngine._aminos
+        assert current_resname in PolymerProposalEngine._aminos
+
+        new_rescharge, current_rescharge = 0,0
+        resname_to_charge = {current_resname: 0, new_resname: 0}
+        for resname in [new_resname, current_resname]:
+            if resname in PolymerProposalEngine._negative_aminos:
+                resname_to_charge[resname] -= 1
+            elif resname in PolymerProposalEngine._positive_aminos:
+                resname_to_charge[resname] += 1
+
+        return resname_to_charge[new_resname] - resname_to_charge[current_resname]
+
+    @staticmethod
+    def get_counterion_indices(charge_diff,
```

Comment:
Can we call `_get_charge_difference` inside this method instead of providing the `charge_diff`? Like is `old_res` and `new_res` here different from the ones that will be specified in the `current_resname` and `new_resname`?

Reply:
`old_res` is a residue object, whereas `current_resname` is a name.
I don't see a significant reason why its better to call `_get_charge_difference` inside `get_water_indices` over the way it is currently. If you disagree, feel free to modify and I'll approve it.

Reply:
That makes sense!